### ISSUE TITLE
AWX Office Hours Agenda - Feb 14 2023

### ISSUE THREAD

**Issue by john-westcott-iv** [ansible/awx#13431]
# AWX Office Hours

## Proposed agenda based on topics

* @erz4: liveness and readiness prob for awx containers https://github.com/ansible/awx-operator/pull/1188
* @stanislav-zaprudskiy graceful termination of awx pods https://github.com/ansible/awx-operator/pull/1193 and https://github.com/ansible/awx/pull/13445
* @TheRealHaoLiu receptor saga part 2: the horror of mismatched k8s API server and node version https://github.com/ansible/awx/issues/13469 and https://github.com/ansible/receptor/pull/729
* @fosterseth tagging awx-ee images with the AWX release https://github.com/ansible/awx/pull/13338
* @relrod https://github.com/ansible/awx-ee/pull/160 and https://github.com/ansible/awx-ee/pull/165
* @miles-w-3 https://github.com/ansible/awx-operator/pull/1183 fix to include previous version of helm chart in helm index
* @mani3887: Can we use redis to help resume job after awx controlplane pod restart
* @nickjmv:  How to enable SAML via the extra_settings instead of the GUI.
* @TheRealHaoLiu first time contributor Spud light
  * Congratulations to https://github.com/OscarBell for his first contribution to the AWX project https://github.com/ansible/awx/pull/13379
  * Congratulations to https://github.com/OscarBell for his first contribution to the AWX project https://github.com/ansible/awx/pull/13379
* @TheRealHaoLiu announcing the recipient of the "Angry Spud" award
  *  Congratulations to @philipsd6 for having the [oldest open PR](https://github.com/ansible/awx/pull/4171) in the repo that @AlanCoding will eventually get to.

## What

After a successful Contributor Summit in October, one of the bits of feedback we got was to host a regular time for the Automation Controller (AWX) Team to be available for your folks in the AWX Community, so we are happy to announce a new regular video meeting.

This kind of feedback loop is vital to the success of AWX and the AWX team wants to make it as easy as possible for you - our community - to get involved.

## Where & When
Our next meeting will be held on Tuesday, Feb 14th, 2023 at [1500 UTC](https://dateful.com/time-zone-converter?t=15:00&tz=UTC)

* [Google Meet](https://meet.google.com/vyk-dfow-cfi)
* Via Phone PIN: 842522378 [Guide](https://support.google.com/meet/answer/9518557)

This meeting is held once a month, on the second Tuesday of the month, at [1500 UTC](https://dateful.com/time-zone-converter?t=15:00&tz=UTC)

## How

Add one topic per comment in this GitHub issue

If you don't have a GitHub account, jump on [#awx:ansible.com](https://matrix.to/#/#awx:ansible.com) on Matrix and we can add the topic for you

## Talk with us

As well as the fortnightly video meeting you can join the Community (inc development team) on Matrix Chat.

* Matrix: [#awx:ansible.com](https://matrix.to/#/#awx:ansible.com) (recomended)
* libera.chat IRC: `#ansible-awx` (If you are already setup on IRC)

The Matrix & IRC channels are bridged, you'll just have a better experience on Matrix

## Links
[AWX YouTube Chanel](https://www.youtube.com/@ansible-awx)
[Previous Meeting](https://github.com/ansible/awx/issues/13325)
[Meeting recording](https://www.youtube.com/watch?v=s_GFbsMCfgY)
[Next Meeting](https://github.com/ansible/awx/issues/13563)

See you soon!

**Comment by john-westcott-iv** [comment]
From @mani3887:

Hi Folks,

Thanks for the AWX office hours. I went through the recording and other videos listed in Ansible AWX community. The videos were helpful in giving clarity of some of the designs followed in AWX. We are running AWX version 21.4 in AKS cluster. At our application we observe a below issue:

When AWX control pod restarts, it seems like it abandons the previous control pod's AWX EE pods, i.e. it seems like the new AWX control pod does not detect the already running AWX EE pods. Would it help if we move the redis cache from in-memory to cloud(azure redis cache)? Is there support for this in AWX? If there is, if we have multiple AWX replicas, and all having redis cache in cloud, would there be sync issues across the AWX instances? I understand that if we did move to a cloud-based solution from in-memory, the latency would be higher.
We had the below questions:

Redis cache - In the videos, it has been mentioned that in-memory redis is primarily used for dumping the logs, dispatcher gets the logs from EE pods, sends it to redis cache, the callback handler gets this from the redis cache and dumps it in the DB. Is the redis cache used for any other functionality? Is this only for the ansible job output logs or it also does the same for the metadata of the job? In our application, we are currently re-directing the Ansible logs(executed on EE pod) to a mounted file share; would the AWX still gather the logs and forward it to dispatcher? we have a solution for transferring the logs, but ideally we would want the metadata for the logs as well.
Role of AWX task container especially during the spinning up of the new pods; does it reconcile with the DB and then spin up the pod; while viewing the youtube videos, it seemed like when we hit the launch button on the UI, a pod would move to Pending state, my question is: would this information also be stored in the DB?
Thanks in advance.
Regards,
Mani

**Comment by john-westcott-iv** [comment]
From @nickjmv:

A tutorial on how to enable SAML via the extra_settings instead of the GUI.

This would (probably) be a solution for this: https://github.com/ansible/awx/issues/12238

**Comment by erz4** [comment]
We want to advance that topic (pr) we encounter on issues when the deployment is rolled out
https://github.com/ansible/awx-operator/pull/1188

**Comment by TheRealHaoLiu** [comment]
congratulations to https://github.com/mahaffey for his first contribution to the AWX project https://github.com/ansible/awx/pull/13483

**Comment by TheRealHaoLiu** [comment]
congratulations to https://github.com/OscarBell for his first contribution to the AWX project https://github.com/ansible/awx/pull/13379

**Comment by stanislav-zaprudskiy** [comment]
Let's discuss https://github.com/ansible/awx-operator/pull/1193 (and https://github.com/ansible/awx/pull/13445) during the meeting

**Comment by TheRealHaoLiu** [comment]
receptor saga pt2 - the horror of mismatched k8s API server and node version

https://github.com/ansible/awx/issues/13469

https://github.com/ansible/receptor/pull/729

TL;DR: `RECEPTOR_KUBE_SUPPORT_RECONNECT` now disable by default

please only re-enable after checking node version

**Comment by TheRealHaoLiu** [comment]
receptor saga pt3 - k8s client side rate limiting (its too late in the day for me to come up with a clever name)

https://github.com/ansible/receptor/pull/735

**Comment by miles-w-3** [comment]
Topic: 
Walk through resolution of helm index - 
- Build index file each release and push it to gh-pages branch
- Now just use helm-index instead of cr
Discuss limitations of helm index -
- solution is built from scratch instead of using wide
  - necessary because the fully-built chart is not checked into git, so the cr pipeline does not detect differences on each release
- helm-index overhead
  - requires downloading all previous releases in order to hash them and put them in the index file
  - dates are not accurate in the index file, since they generate
- Potential improvement
  - use yq or something similar to see which releases are already in index yaml, only add new release
  - This would lessen overhead, and not increment all dates on each release
  - This would leverage the merge flag in helm index which currently isn't used, since the command regenerates and overwrites conflicting entries in the yaml file

**Comment by TheRealHaoLiu** [comment]
@relrod talk about new awx-ee stream9 base image

**Comment by AlanCoding** [comment]
A candidate patch for https://github.com/ansible/ansible-runner/issues/1187 is up, and that will affect a good number of people who hit this kind of error.
